engine/schema/src/org/apache/cloudstack/resourcedetail/ResourceDetailsDao.java

```diff
@@ -77,4 +77,6 @@
 
     public void addDetail(long resourceId, String key, String value, boolean display);
 
+    public void removeDetails(String key);
```

Comment:
Again what is the need for this? There are already removeDetail/removeDetails() method present.